pkg/monitor/monitor/monitor_manager.go

```diff
@@ -427,3 +442,96 @@ func (m *MonitorManager) patchTidbClusterStatus(tcRef *v1alpha1.TidbClusterRef,
 	_, err = m.deps.Clientset.PingcapV1alpha1().TidbClusters(tc.Namespace).Patch(tc.Name, types.MergePatchType, mergePatch)
 	return err
 }
+
+func (m *MonitorManager) smoothMigrationToStatefulSet(monitor *v1alpha1.TidbMonitor) (bool, error) {
```

Comment:
What about the following logic:
Get deployment successfully, delete it and return
Get deployment error except for the non NotFound error, return
Get deployment NotFound error
    PVC found
        PV not found, return true, no need to migrate
        PV found
            patch the status of monitor to make sure the PV info is updated (return error for failure)
            delete PVC
            patch PV
    PVC not found
        no PV in status of monitor, return true, no need to migrate
        PV found in status of monitor, patch PV (only patch if necessary, check the claimRef first)

Reply:
There are too many nested `if`s, please optimize the code to handle the code where it can return quickly first.